server/src/main/java/org/opensearch/plugins/SearchPipelinesPlugin.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.search.pipeline.Processor;
+
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * An extension point for {@link Plugin} implementation to add custom search pipeline processors.
+ *
+ * @opensearch.api
+ */
+public interface SearchPipelinesPlugin {
```

Comment:
```suggestion
public interface SearchPipelinePlugin {
```

Super minor, I we use `SearchPipelineXxx` everywhere, the plural form here is a bit unexpected

Reply:
:+1: